### DISCUSSION TITLE
Open In KItty option in the context menu

### DISCUSSION THREAD

**Discussion by AyushSehrawat** [kovidgoyal/kitty#6038] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
I want to add an option in the context menu ( which can be opened when you go to the file manager and right click ) which says `Open In Kitty` ( similar to Open in Terminal option shown by default ).

**Describe the solution you'd like**
A way to do it. I know it's not a kitty related feature request , rather ubuntu related, but i didn't find anything about it.
What i want - Able to show Open in Kitty which opens kitty in the directory of where i clicked , like i go to `~/.config` in Files ( file manager ) and right click , it shows `Open in Kitty` , clicking it opens kitty in the same directory ( i.e `~/.config` )

**Describe alternatives you've considered**
NIL

**Additional context**
NIL

**Comment by kovidgoyal** [comment]
You should ask for help in a ubuntu forum. Or post in the discussions. THis is not an issue.

**Reply by AyushSehrawat** [reply]
> You should ask for help in a ubuntu forum. Or post in the discussions. THis is not an issue.

Ah sorry , i forgot about discussions. Sorry for creating this an issue. 

I will ask in ubuntu forum.

**Comment by page-down** [comment, marked as answer]
> ... add an option in the context menu ... file manager ...

You didn't mention which file manager and its version, so I assume it's the "Files" (nautilus) under latest GNOME.

Before GNOME 43, you could customize actions using `filemanager-actions`, a package that exists in Ubuntu 20.04. After that it was gone, and the upstream git repository was archived and will not be updated in any way.
Apparently GNOME does not want you to be able to freely configure the software and see this as an **improvement**.

> ... similar to Open in Terminal option shown by default ...

This is an extension from `gnome-terminal`.
As far as I know there is no well-known extension to customize this. You can find an existing one to learn from, and use the `libnautilus-extension` lib from `nautilus` to implement yours.

The `Nautilus script` feature also seems to be working in the latest version.
You can write a script that parses the paths in the environment variables and opens them using a new instance of kitty (`kitty +open`, `kitty --working-directory`) or via remote control.

You can also write extensions in Python.
- Install the package `nautilus-python`. (`python3-nautilus` on Ubuntu 22.10)
- Get the `open-terminal.py` example script.
    - v4.0: https://gitlab.gnome.org/GNOME/nautilus-python/-/blob/4.0/examples/open-terminal.py
- Modify the program to be executed, the default is `gnome-terminal`. I would suggest using kitty remote control (via socket) to open new OS windows, tabs, windows. If you run `kitty` right there, `Files` will hang until kitty exits.
    - `kitten @ --to unix:/tmp/mykitty launch --cwd /path/to/dir`

For KDE Dolphin, see my answer at the following link.
https://github.com/kovidgoyal/kitty/discussions/4233#discussioncomment-1636048

**Reply by AyushSehrawat** [reply]
I ended up using https://github.com/Stunkymonkey/nautilus-open-any-terminal.

It was somehow already there in nautilus-python directory ( maybe default or with some program)..
So i edited the name in dconf editor to set it to kitty. And it was done!

Btw thanks for your help
